pkg/daemon/update.go

```diff
@@ -423,10 +427,32 @@ func (dn *CoreOSDaemon) applyOSChanges(mcDiff machineConfigDiff, oldConfig, newC
 	}
 
 	if dn.nodeWriter != nil {
-		dn.nodeWriter.Eventf(corev1.EventTypeNormal, "OSUpdateStaged", "Changes to OS staged")
+		// We send out the event OSUpdateStaged synchronously to ensure it is recorded.
+		// Remove this when we can ensure all events are sent before exiting.
+		ref, err := reference.GetReference(scheme.Scheme, dn.node)
+		if err == nil {
+			t := metav1.Time{Time: time.Now()}
+			event := &corev1.Event{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("%v.%x", ref.Name, t.UnixNano()),
+					Namespace: metav1.NamespaceDefault,
+				},
+				Reason:         "OSUpdateStaged",
+				Type:           corev1.EventTypeNormal,
+				Message:        "Changes to OS staged",
+				FirstTimestamp: t,
+				LastTimestamp:  t,
+				Count:          1,
+				Source:         corev1.EventSource{Component: "machineconfigdaemon", Host: dn.name},
+			}
+			// its ok to create a unique event for this low volume event
+			if _, err = dn.kubeClient.CoreV1().Events(metav1.NamespaceDefault).Create(context.TODO(),
+				event, metav1.CreateOptions{}); err != nil {
+				utilruntime.HandleError(err)
```

Comment:
Hmm, does this match existing error behaviour? i.e. if our event fails to fire in the current daemon, do we fail the update loops?

Reply:
No, it doesn't match existing behaviour. Removed and add logging instead.